processor/transformer/transformer.go

```diff
@@ -270,7 +285,13 @@ func (trans *handle) Transform(ctx context.Context, clientEvents []TransformerEv
 
 // UserTransform function is used to invoke user transformer API
 func (trans *handle) UserTransform(ctx context.Context, clientEvents []TransformerEvent, batchSize int) Response {
-	return trans.transform(ctx, clientEvents, trans.userTransformURL(), batchSize, userTransformerStage)
+	var dehydratedClientEvents []TransformerEvent
+	for _, clientEvent := range clientEvents {
+		dehydratedClientEvent := clientEvent.Dehydrate()
```

Comment:
What is a dehydrated client in this context? :thinking:

Reply:
Added a little more in the description. Basically we send a common payload across user transformation , destination transformation and tracking plan which heavily inflates the payload size for each of them. The idea is to reduce the payload size by removing the unnecessary fields

Reply:
Whoever comes across this code will still have troubles understanding what `Dehydrate` means though, right? I don't think they'll go chasing the PR where you created this to get more context.

I suggest adding a comment here and also renaming the `Dehydrate` method to better reflect what it does:
1. PruneFields
2. StripUnnecessaryFields
3. GetVersionsOnly
4. ...